python/lsst/verify/compatibility/metricsControllerTask.py

```diff
@@ -0,0 +1,420 @@
+            inputDataIds[param] = [ref.dataId for ref in inputRefs]
+
+        outputDataIds = {"measurement": dataref.dataId}
+        try:
+            result = metricTask.adaptArgsAndRun(inputData, inputDataIds,
+                                                outputDataIds)
+            value = result.measurement
+            if value is not None:
+                job.measurements.insert(value)
+        except MetricComputationError:
+            # Apparently lsst.log doesn't have built-in exception support?
+            self.log.error("Measurement of %r failed on %s->%s\n%s",
+                           metricTask, inputDataIds, outputDataIds,
+                           traceback.format_exc())
+
+    def runDataRefs(self, datarefs):
+        """Calls all registered metric tasks on each dataref.
+
+        This method shall load all datasets required to compute a particular
+        metric, and shall persist the metrics as one or more `lsst.verify.Job`
+        objects. Some objects may be persisted, or incorrectly persisted, in
+        the event of an exception.
+
+        Parameters
+        ----------
+        datarefs : `list` of `lsst.daf.persistence.ButlerDataRef`
+            The data to measure. Datarefs may be complete or partial; each
+            shall generate a measurement at the same granularity (e.g., a
+            dataref with only `"visit"` specified shall generate
+            visit-level measurements).
+
+        Returns
+        -------
+        struct : `lsst.pipe.base.Struct`
+            A Struct containing at least the following component:
+
+            - ``jobs`` : a `list` of collections of measurements
+              (`lsst.verify.Job`). Each job in the list shall contain the
+              measurement(s) for the corresponding dataref, and each job
+              shall have up to one measurement for each element
+              in `self.measurers`. A particular measurement shall be omitted
+              if it could not be created.
+        """
+        jobs = []
+        index = 0
+        for dataref in datarefs:
+            job = Job.load_metrics_package()
+            try:
+                # TODO: generalize this in DM-16642
+                job.meta['instrument'] = \
+                    MetricsControllerTask._getInstrument(dataref)
+                job.meta.update(dataref.dataId)
+
+                for task in self.measurers:
+                    self._computeSingleMeasurement(job, task, dataref)
+            finally:
+                jobFile = self._nameJobFile(index, dataref.dataId)
+                self.log.info("Persisting metrics to %s...", jobFile)
+                # This call order maximizes the chance that job will be
+                # written, and to a unique file
+                job.write(jobFile)
+                index += 1
+                jobs.append(job)
+
+        return Struct(jobs=jobs)
+
+    def _nameJobFile(self, index, dataId):
```

Comment:
Normally when we say "file" in Python we're talking about a file handle. I think in this case its a "Path" instead?

Reply:
"path" seems kind of redundant with "name"; would `_jobFilePath` be acceptable?